index.js

```diff
@@ -1,21 +1,35 @@
-// Inspired by (but complete rewrite of) requirejs-metagen
-let _ = require('lodash/fp')
-let Promise = require('bluebird')
-let readDir = Promise.promisify(require('recursive-readdir'))
-let fs = Promise.promisifyAll(require('fs'))
-
-// Path Utils
-let filesRelative = (dir, ex) => readDir(dir, ex).map(x => x.slice(dir.length))
+// Inspired by (but complete rewrite of) requirejs-metagen.
+let _       = require('lodash/fp') // Add lodash helper functions.
+let Promise = require('bluebird')  // Replace promises with the Bluebird Promises library. ("Bluebird cuts corners" -- yikes!)
+let readDir = Promise.promisify(require('recursive-readdir')) // Lists all files in a certain directory and all its sub-directories.
+let fs      = Promise.promisifyAll(require('fs')) // Add the filesystem library.
+
+// Define path Utils.
+
+// Get all the file names in a certain directory and its sub-directories, excluding the directory path.
+let filesRelative = (dir, exclusions) => readDir(dir, exclusions).map(filename => filename.slice(dir.length))
+
+// Get a file's name without its extension.
```

Comment:
If you think this comment is needed, then we should have named the method better as something like `stripExtension` or similar. In our `futil-js` library, we go a bit overboard with docs but that's because all the methods are publicly exposed. Ideally code should be self documenting and comments should only be there if they are actually adding value.